tailormap-components/projects/core/src/lib/user-interface/attributelist/layer.service.ts

```diff
@@ -24,11 +23,43 @@ export class LayerService {
               private highlightService: HighlightService) {
     // Install the layerVisibilityChanged handler.
     this.tailorMapService.layerVisibilityChanged$.subscribe(value => {
-      this.loadLayers();
+      // layerVisibilityChanged visible to true occurs too often (also if layer is already visible)
+      // console.log ('LayerService visi changed value: ' + value.visible);
+      if (value.visible) {
+        if (!this.isLayerIdInLayers(value.layer.id)) {
+          this.addLayer(value.layer.id);
+        }
+      } else {
+        if (this.isLayerIdInLayers(value.layer.id)) {
+          this.removeLayer(value.layer.id);
+        }
+      }
+      this.layersSubject.next(this.layers);
     });
     this.loadLayers();
   }
 
+  /**
+   * Returns part from full layer name before ":". Converts to lowercase too.
+   */
+  private static sanitizeLayername(layerName: string): string {
+    const index = layerName.indexOf(':');
+    if (index !== -1) {
+      layerName = layerName.substring(index + 1);
+    }
+    return layerName.toLowerCase();
+  }
+
+  private isLayerIdInLayers (layerId): boolean {
+    let layerFound = false;
+    let index = 0;
+    while (index < this.layers.length && !layerFound) {
+      layerFound = (this.layers[index].id === layerId);
+      index++;
+    }
```

Comment:
Kan je hier niet beter een this.layers.find voor gebruiken?

Reply:
Zal ik doen, ik ben gewend om find te ontwijken vanwege ondersteuning oudere browsers, maar dat is hier mogelijk geen probleem.

Reply:
Ik ben er nog een tegengekomen, getTabIndexByLayerId, dat heb ik met findIndex opgelost.